packages/rum-core/src/domain/tracing/tracer.ts

```diff
@@ -28,22 +28,29 @@ export function startTracer(configuration: Configuration): Tracer {
     clearTracingIfCancelled,
     traceFetch: (context) =>
       injectHeadersIfTracingAllowed(configuration, context, (tracingHeaders: TracingHeaders) => {
-        context.init = { ...context.init }
-        const headers: string[][] = []
-        if (context.init.headers instanceof Headers) {
-          context.init.headers.forEach((value, key) => {
-            headers.push([key, value])
-          })
-        } else if (Array.isArray(context.init.headers)) {
-          context.init.headers.forEach((header) => {
-            headers.push(header)
-          })
-        } else if (context.init.headers) {
-          Object.keys(context.init.headers).forEach((key) => {
-            headers.push([key, (context.init!.headers as Record<string, string>)[key]])
+        if (context.init instanceof Request) {
```

Comment:
`context.init` match `fetch` optional second parameter:

```ts
fetch(input: RequestInfo, init?: RequestInit)

type RequestInfo = Request | string
```

FMU, this condition will never be true IRL